gradle-settings-conventions/src/main/kotlin/customization.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import com.gradle.develocity.agent.gradle.DevelocityConfiguration
+import org.gradle.api.initialization.Settings
+
+fun Settings.enrichTeamCityData() {
+    val ge = extensions.getByType(DevelocityConfiguration::class.java)
+
+    gradle.projectsEvaluated {
+        if (isCIRun) {
+            val buildTypeId = "teamcity.buildType.id"
+            val buildId = "teamcity.build.id"
+
+            if (gradle.rootProject.hasProperty(buildId) && gradle.rootProject.hasProperty(buildTypeId)) {
+                val buildIdValue = gradle.rootProject.property(buildId).toString()
+                val teamCityBuildNumber = java.net.URLEncoder.encode(buildIdValue, "UTF-8")
+                val teamCityBuildTypeId = gradle.rootProject.property(buildTypeId)
+
+                ge.buildScan.link(
+                    "Ktor TeamCity build",
+                    "${TEAMCITY_URL}/buildConfiguration/${teamCityBuildTypeId}/${teamCityBuildNumber}"
+                )
+            }
+
+            if (gradle.rootProject.hasProperty(buildId)) {
+                ge.buildScan.value("CI build id", gradle.rootProject.property(buildId) as String)
+            }
+        }
+
+        if (!isCIRun) {
+            // Git commit id
+            val commitId = execute("git rev-parse --verify HEAD")
```

Comment:
**Nit:** Maybe we should move it to a separate function with a different name? Like `enrichGitData`. Or rename the current function.

Reply:
Sure, that'd be better